userspace/libsinsp/sinsp.h

```diff
@@ -850,148 +851,30 @@ class SINSP_PUBLIC sinsp : public capture_stats_source
 
 	/*!
 		\brief Mark desired syscall as (un)interesting, enabling or disabling its collection.
-		This method receives a `ppm_sc` code, not a syscall system code, the same ppm_code
-		can match more than one system syscall. You can find the available
-		`enum ppm_syscall_code` in `driver/ppm_events_public.h`.
-		Please note that this method must be called when the inspector is already open to 
+		Note that the same ppm_code can match multiple system syscalls.
+
+		Please note that this method must be called when the inspector is already open to
 		modify at runtime the interesting syscall set.
 
 		WARNING: playing with this API could break `libsinsp` state collection, this is only
 		useful in advanced cases where the client needs to know what it is doing!
 	*/
-	void mark_ppm_sc_of_interest(uint32_t ppm_sc, bool enabled = true);
-
-	/*!
-		\brief Provide the minimum set of syscalls required by `libsinsp` state collection.
-		If you call it without arguments it returns a new set with just these syscalls
-		otherwise, it merges the minimum set of syscalls with the one you provided.
-
-		WARNING: without using this method, we cannot guarantee that `libsinsp` state
-		will always be up to date, or even work at all.
-	*/
-	std::unordered_set<uint32_t> enforce_sinsp_state_ppm_sc(std::unordered_set<uint32_t> ppm_sc_of_interest = {});
```

Comment:
Moving all these is the biggest breaking change of this PR. Please let's make sure the commit names and release notes have a conventional commit prefix containing `!`, and let's be explicit about the action items for fixing the issues of whoever pulls this upstream!

Reply:
You are right, i will review each commit and add the `!` where needed ;)

Reply:
Done!